data-processing/common/parse_equation.py

```diff
@@ -64,23 +66,54 @@ def get_symbols(mathml: str) -> List[Symbol]:
             if index is not None:
                 characters.append(index)
 
-        # Clean the MathML of our annotations. Do this cleaning on a clone of the node, so that the
-        # annotations are available for processing later nodes.
-        node_clone = copy.copy(symbol_node)
-        _remove_s2_annotations(node_clone)
-        for descendant in node_clone.descendants:
-            _remove_s2_annotations(descendant)
+        # Group consecutive character symbols only if they are a child of the mrow tag.
+        # The .has_attr('s2:end') is necessary as not every symbol is annotated by katex
+        if symbol_node.has_attr('s2:end') and symbol_node.parent.name in MERGABLE_PARENT_TAGS and symbol_node.name in CHARACTER_TAGS:
+            if len(consecutive_chars) > 0 and not consecutive_chars[-1]['s2:end'] == symbol_node['s2:start']:
+                node_symbols.append(_merge_consecutive_symbols(consecutive_chars, all_indices, soup.new_tag('mi')))
+                consecutive_chars = []
+                all_indices = []
+            consecutive_chars.append(symbol_node)
+            all_indices.extend(characters)
+        elif len(consecutive_chars) > 0:
+            node_symbols.append(_merge_consecutive_symbols(consecutive_chars, all_indices, soup.new_tag('mi')))
+            consecutive_chars = []
+            all_indices = []
+
+        if len(consecutive_chars) == 0:
```

Comment:
I'm also confused about when this condition gets triggered. Can you give the condition some type of descriptive variable name, or provide some comment?

Reply:
Yeah I'm going to rewrite this statement as a whole. Basically there are four cases that need to be addressed (realizing now that this can just be put into two cases):
1. The symbol we are one is part of a consecutive sequence of symbols 
    1a) that consecutive sequence is our current consecutive sequence.
     1b) we had a consecutive sequence, but this one is part of a different consecutive sequence. 
2. The symbol is not part of a consecutive sequence
   2a) We were building a consecutive sequence and thus now need to add it in before we build our current symbol
   2b) We were not building a consecutive sequence, business as usual.